src/constructs/core/parameters.ts

```diff
@@ -86,3 +86,13 @@ export class GuAmiParameter extends GuParameter {
     });
   }
 }
+
+export class GuArnParameter extends GuStringParameter {
+  constructor(scope: GuStack, id: string, props: GuNoTypeParameterProps) {
+    super(scope, id, {
+      ...props,
+      allowedPattern: "arn:aws:[a-z0-9]*:[a-z0-9\\-]*:[0-9]{12}:.*",
```

Comment:
Could we add some unit tests against the regex? We could pull it out into a const somewhere and test it in it's own right as a regex pattern?

Reply:
Good shout! I've just added these, and discovered that we'll need a separate parameter regex for S3 ARNs, so I've added that!